test/src/single_thread_matrix_calculation_test.cpp

```diff
@@ -161,14 +161,44 @@ TEST_F(TestSinglThreadCalculation, subtractNumberSubMatrix) {
     Matrix<> result2({{-1, -1, -1}, {7, 6, 5}, {3, 3, 3}});
     ASSERT_TRUE(equalSingleThread(output2, result2, 0, 0, {3, 3}));
 }
-// TODO
-TEST_F(TestSinglThreadCalculation, multiplyNumberWholeMatrix) {}
-// TODO
-TEST_F(TestSinglThreadCalculation, multiplyNumberSubMatrix) {}
-// TODO
-TEST_F(TestSinglThreadCalculation, divideNumberWholeMatrix) {}
-// TODO
-TEST_F(TestSinglThreadCalculation, divideNumberSubMatrix) {}
+
+TEST_F(TestSinglThreadCalculation, multiplyNumberWholeMatrix) {
+    output = Matrix<>({3, 3}, -1);
+    multiplySingleThread(a, b, output, 0, 0, a.getShape());
+    Matrix<> result({{0, 0, 0}, {0, 1, 0}, {0, 0, 2}});
+    ASSERT_TRUE(equalSingleThread(output, result, 0, 0, output.getShape()));
+}
+
+TEST_F(TestSinglThreadCalculation, multiplyNumberSubMatrix) {
+    output = Matrix<>({3, 3}, -1);
+    multiplySingleThread(a, b, output, 1, 0, {2, 3});
+    Matrix<> result({{-1, -1, -1}, {0, 1, 0}, {0, 0, 2}});
+    ASSERT_TRUE(equalSingleThread(output, result, 0, 0, output.getShape()));
+}
+
+TEST_F(TestSinglThreadCalculation, divideNumberWholeMatrix) {
```

Comment:
There are two types of divideNumber tests:
* divideSinglethread(number, a,...)
* divideSingleThread(a, number,...)

Reply:
Remove the period of the title of the commit

Add bullet points